var/spack/repos/builtin/packages/care/package.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Care(CMakePackage, CudaPackage, HipPackage):
+    """
+    Algorithms for chai managed arrays.
+    """
+
+    homepage = "https://github.com/LLNL/CARE"
+    git      = "https://github.com/LLNL/CARE.git"
+
+    version('develop', branch='develop', submodules='True')
+    version('master', branch='main', submodules='True')
+    version('0.3.0', tag='v0.3.0', submodules='True')
+    version('0.2.0', tag='v0.2.0', submodules='True')
+
+    variant('openmp', default=False, description='Build Shared Libs')
+    variant('implicit_conversions', default=True, description='Enable implicit'
+            'conversions to/from raw pointers')
+    variant('benchmarks', default=True, description='Build benchmarks.')
+    variant('examples', default=True, description='Build examples.')
+    variant('docs', default=False, description='Build documentation')
+
+    depends_on('blt', type='build')
+
+    depends_on('camp')
+    depends_on('umpire@develop')
+    depends_on('raja@develop')
+    depends_on('chai@develop+enable_pick~benchmarks')
+
+    depends_on('camp+cuda', when='+cuda')
+    depends_on('umpire@develop+cuda', when='+cuda')
+    depends_on('raja@develop+cuda', when='+cuda')
+    depends_on('chai@develop+cuda+enable_pick~benchmarks', when='+cuda')
+
+    # TODO: support external cub rather than in-package submodule.
+    # depends_on('cub', when='+cuda')
```

Comment:
This scares me -- if this is used as a dependency by anything that uses an external cub, we will generate a loader race in the resulting code.

Reply:
Removed for the time being. Fixing/Testing this stuff with cuda should be a separate pull request

Reply:
The TODO wasn't so much the part that scared me, the fact it uses an internal `cub` scares me. Can we add a very obvious warning comment in the package that it currently only supports internal `cub` and will have a loader race when compiled with anything else using `cub`?

Reply:
Added a clearer warning